### DISCUSSION TITLE
Using API rules to restrict only "pro" members? Possible?

### DISCUSSION THREAD

**Discussion by fayazara** [pocketbase/pocketbase#3744] [Q&A]
Hello,

I am currently working on an article and creating a YouTube tutorial focused on building a small full-stack demo application using Pocketbase, Nuxt, and Stripe for subscriptions.

I have successfully implemented the basic functionalities of the app, as demonstrated in the demo below. You can also access the repository here: [https://github.com/fayazara/pocketbase-nuxt](https://github.com/fayazara/pocketbase-nuxt), still WIP I will refactor once I get the initial idea working.

https://github.com/pocketbase/pocketbase/assets/15716057/f0874c38-c490-43ca-acae-015fe5b5822e

My next goal is to integrate Stripe subscriptions. Essentially, when a user purchases a plan, I intend to add an `isProMember` flag to the user table and set it to `true`. While I can certainly implement this check within my code, I'm curious to know if it's possible to handle this via an API rule.

I'm struggling with constructing a rule for the "Create" action. Specifically, I want to add this rule, 

## What I want to do
**_Free members can only create three posts and pro members can create unlimited posts_** 

something like -  `@request.auth.id = owner.id` and if `@owner.isProMember` is `true`, also ensuring that `@owner.planExpiryDate` is not past the current date. Any guidance on how to formulate this rule would be greatly appreciated. Perhaps @ganigeorgiev could offer some insight?

Here's my collection schema for reference:
![Screenshot 13-11 at 18](https://github.com/pocketbase/pocketbase/assets/15716057/6fe0e0db-7272-4b2d-bf90-e32a0324c0b7)

Thank you in advance for your help!

**Comment by ganigeorgiev** [comment]
> I'm struggling with constructing a rule for the "Create" action

Create rule for the "posts" table? In that case using something like `@request.auth.isProMember = true` should work but I don't understand what is the desired behavior and constraints.

In any case, note that to prevent users from changing their own `stripeId`, `isProMember`, `planExpiryDate`, etc. fields you'll have to define create/update API rules for the "users" collection like `@request.data.isProMember:isset = false && @request.data.planExpireyDate:isset = false && ...`, etc.
An alternative approach is to move the "private" fields to a new collection (eg. "payments") with a relation to "users" that could be edited only by admins.

**Reply by fayazara** [reply]
That's a good catch, I will move all of my payment related fields to a new table

**Comment by fayazara** [comment]
I constructed the rule to only allow pro members - `@request.auth.id = owner.id && @request.auth.isProMember = true && @request.auth.planExpiryDate >= @todayStart`. 

I guess `@collection.count` is not possible currently

**Reply by ganigeorgiev** [reply]
> I guess `@collection.count is not possible currently

I'm not sure I follow. If you need to count the collection records, you can create a helper [View collection](https://pocketbase.io/docs/collections/#view-collection) with the specific aggregation you need and then "join" it  in your main collection rule with `@collection.yourView.someTotalField > 10`.

**Reply by fayazara** [reply]
Okay,I got some idea of what to do, let me try to implement this.
